### DISCUSSION TITLE
Writing parser on macOS M1

### DISCUSSION THREAD

**Discussion by gyzerok** [tree-sitter/tree-sitter#1959] [Q&A] [answered]
Hello everyone!

I am trying to create my first parser with tree-sitter. For the grammar I use https://github.com/projectfluent/tree-sitter-fluent/blob/master/grammar.js

When I do `tree-sitter parse example-file` I get the following error, which I suspect has something to do with my mac running on M1.

Tried with both homebrew installation and npm one. Can someone help me?

```
Failed to load language in current directory

Caused by:
    Parser compilation failed.
    Stdout: 
    Stderr: Undefined symbols for architecture x86_64:
      "_tree_sitter_fluent_external_scanner_create", referenced from:
          _tree_sitter_fluent.language in parser-305e9a.o
      "_tree_sitter_fluent_external_scanner_deserialize", referenced from:
          _tree_sitter_fluent.language in parser-305e9a.o
      "_tree_sitter_fluent_external_scanner_destroy", referenced from:
          _tree_sitter_fluent.language in parser-305e9a.o
      "_tree_sitter_fluent_external_scanner_scan", referenced from:
          _tree_sitter_fluent.language in parser-305e9a.o
      "_tree_sitter_fluent_external_scanner_serialize", referenced from:
          _tree_sitter_fluent.language in parser-305e9a.o
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by amaanq** [comment, marked as answer]
The error itself is not M1-Mac specific. That looks like you've defined an external token in the grammar, but there's no corresponding scanner.c. Though in the linked grammar, I don't see any external tokens or scanner, so I'm going to assume this is either outdated or solved. Feel free to reply if I'm wrong though.
